### PR TITLE
bael-4793: add main source

### PR DESCRIPTION
**Pull request by hmdrzsharifi**

### CODE REVIEW

File: spring-boot-rest-2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.baeldung.web</groupId>
+    <artifactId>spring-boot-rest-2</artifactId>
+    <name>spring-boot-rest</name>
```
Comment (KevinGilmore) [inline spring-boot-rest-2/pom.xml:8]:
```suggestion
    <name>spring-boot-rest-2</name>
```

Comment (KevinGilmore) [reply spring-boot-rest-2/pom.xml:8]: name is optional but, if present, must match artifactId

File: spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/AnnotationDrivenEndpointsListener.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@Configuration
+public class AnnotationDrivenEndpointsListener {
+    private final Logger LOGGER = LoggerFactory.getLogger("AnnotationDrivenEndpointsListener.class");
+
+    @EventListener
+    public void handleContextRefresh(ContextRefreshedEvent event) {
+        ApplicationContext applicationContext = event.getApplicationContext();
+        RequestMappingHandlerMapping requestMappingHandlerMapping = applicationContext
+                .getBean("requestMappingHandlerMapping", RequestMappingHandlerMapping.class);
```
Comment (KevinGilmore) [inline spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/AnnotationDrivenEndpointsListener.java:23]: indent only two spaces from where previous line begins

File: spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/EndpointsListener.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+@Configuration
+public class EndpointsListener implements ApplicationListener<ContextRefreshedEvent> {
+    private final Logger LOGGER = LoggerFactory.getLogger("EndpointsListener.class");
+
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        ApplicationContext applicationContext = event.getApplicationContext();
+        RequestMappingHandlerMapping requestMappingHandlerMapping = applicationContext
+                .getBean("requestMappingHandlerMapping", RequestMappingHandlerMapping.class);
```
Comment (KevinGilmore) [inline spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/EndpointsListener.java:23]: indentation

File: spring-boot-rest-2/src/main/java/com/baeldung/endpoint/swagger/SpringFoxConfig.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+@Configuration
+public class SpringFoxConfig {
+
+    @Bean
+    public Docket api() {
+        return new Docket(DocumentationType.SWAGGER_2)
+                .select()
+                .apis(RequestHandlerSelectors.any())
+                .paths(PathSelectors.any())
+                .build();
```
Comment (KevinGilmore) [inline spring-boot-rest-2/src/main/java/com/baeldung/endpoint/swagger/SpringFoxConfig.java:19]: indentation

**Review by KevinGilmore — COMMENTED** [review]
Please fix indentation of continuation lines as noted in several places.

File: spring-boot-rest-2/pom.xml (hunk 1 above)
Comment (hmdrzsharifi) [reply spring-boot-rest-2/pom.xml]: Done

File: spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/AnnotationDrivenEndpointsListener.java (hunk 2 above)
Comment (hmdrzsharifi) [reply spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/AnnotationDrivenEndpointsListener.java]: Done

File: spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/EndpointsListener.java (hunk 3 above)
Comment (hmdrzsharifi) [reply spring-boot-rest-2/src/main/java/com/baeldung/endpoint/listener/EndpointsListener.java]: Done

File: spring-boot-rest-2/src/main/java/com/baeldung/endpoint/swagger/SpringFoxConfig.java (hunk 4 above)
Comment (hmdrzsharifi) [reply spring-boot-rest-2/src/main/java/com/baeldung/endpoint/swagger/SpringFoxConfig.java]: Done

**Pull request merged**